__tests__/config/helpers/localDevnetDetector.ts

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable no-console */
+import { GS_DEFAULT_TEST_PROVIDER_URL } from '../constants';
+import { setIfNullish } from './env';
+
+export type DevnetStrategy = Record<'isDevnet' | 'isRS', boolean>;
+
+const LOCAL_DEVNET_NOT_RUNNING_MESSAGE = `
+Local devnet is not running. In order to properly run it you need to do the following: \n
+  - Go to the: https://hub.docker.com/r/shardlabs/starknet-devnet-rs/tags
+  - Find the latest tag and copy the "docker pull" command
+  - Run the docker on your machine
+  - Run the: "docker pull shardlabs/starknet-devnet-rs:latest"
+`;
+
+class LocalDevnetDetector {
+  private strategy: DevnetStrategy = { isDevnet: false, isRS: false };
+
+  get isDevnet() {
+    return this.strategy.isDevnet;
+  }
+
+  get isRS() {
+    return this.strategy.isRS;
+  }
+
+  private setup() {
+    setIfNullish('IS_LOCALHOST_DEVNET', this.isDevnet);
+    setIfNullish('IS_RPC_DEVNET', this.isDevnet && (this.isRS || !!process.env.TEST_RPC_URL));
+    setIfNullish('IS_SEQUENCER_DEVNET', this.isDevnet && process.env.IS_RPC_DEVNET === 'false');
+    return this.strategy;
+  }
+
+  private async isLocalDevnet(): Promise<boolean> {
```

Comment:
Do we need this check at all? Can we just do the `isRsDevnet` check and if that one fails, just throw the error that your local RS devnet is not running with the following instructions?

Reply:
We need to remove python devnet and sequencer, but my initial plan was to do it in next pr.
Anyway we can do it also in this one.
Yes isLocalDevnet is redundant here

Reply:
I'll remove the `isLocalDevnet`. As for the py devnet and sequencer, we can do that in a separate MR...you can maybe create an issue for the sequencer removal for that one ;)